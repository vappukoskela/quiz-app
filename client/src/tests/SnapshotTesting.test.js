import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SnapshotTesting from '../TestsDemo/SnapshotTesting.js';

// palauttaa aina success - testit toimii
test('this test will run successfully', () => {
    expect(true).toBe(true)
})

// renderöi oletuksena oikein, ei kaadu
it('renders correctly', () => {
    const tree = renderer.create(<SnapshotTesting/>).toJSON()
    expect(tree).toMatchSnapshot();
})

// ottaa propsin ja renderöi vaihtoehtoisen pathin oikein
it('says goodbye', () => {
    const tree = renderer.create(<SnapshotTesting sayGoodbye={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

// etsii <h1> -elementin ja tutkii sitä
it('renders title', () => {
    render(<SnapshotTesting/>);
    const title = screen.getByText('Hello world');
    console.log(title) // pitäisi olla h1-elementti
    expect(title).toMatchSnapshot()
})


