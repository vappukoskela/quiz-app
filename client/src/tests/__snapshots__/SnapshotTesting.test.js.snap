// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Hello world
  </h1>
  nothing to see here!
</div>
`;

exports[`renders title 1`] = `
<h1>
  Hello world
</h1>
`;

exports[`says goodbye 1`] = `
<div>
  <h1>
    Hello world
  </h1>
  goodbye universe
</div>
`;
